views/js/qtiCreator/widgets/interactions/choiceInteraction/states/Question.js

```diff
@@ -101,46 +189,117 @@ define([
+            selectedCase = allowedChoices.find(
+                allowedChoice => allowedChoice.type === type && allowedChoice.constraints === constraints
+            );
+            if (!selectedCase) {
+                resetToSingleNone();
+            }
+            setAttrMaxMinChoices(selectedCase.minChoices, selectedCase.maxChoices);
+        };
+
+        callbacks.type = function (interactionParam, value) {
+            type = value;
+            setSelectedCase();
+            if (type === 'single') {
+                $form.find('[name="constraints"][value="other"]').prop('disabled', true);
+                deleteMinMax();
+            } else {
+                $form.find('[name="constraints"][value="other"]').prop('disabled', false);
+            }
+        };
+
+        callbacks.constraints = function (interactionParam, value) {
+            constraints = value;
+            if (constraints === 'other') {
+                setAttrMaxMinChoices(DEFAULT_MIN, DEFAULT_MAX);
+                createMinMaxComponent(DEFAULT_MIN, DEFAULT_MAX);
+            } else {
+                deleteMinMax();
+                setSelectedCase();
+            }
+        };
+
+        //when the number of choices changes we update the range
+        widget.on('choiceCreated choiceDeleted', function (data, e) {
+            if (data.interaction.serial === interaction.serial) {
+                const choiceCount = _.size(interaction.getChoices());
+                if (
+                    choiceCount <= 1 ||
+                    ($choiceArea.find('.qti-choice:visible').length <= 1 && e.type === 'choiceDeleted')
+                ) {
+                    // multiple choices should be disabled
+                    resetToSingleNone();
+                    $form.find('[name="type"][value="multiple"]').prop('disabled', true);
+                } else {
+                    // multiple choices should be enabled
+                    $form.find('[name="type"][value="multiple"]').prop('disabled', false);
+                }
+                if (constraints === 'other' && minMaxComponent) {
+                    minMaxComponent.updateThresholds(DEFAULT_MIN, choiceCount - 1, 'min');
+                    minMaxComponent.updateThresholds(DEFAULT_MAX, choiceCount, 'max');
+                }
+            }
+        });
+
         formElement.setChangeCallbacks($form, interaction, callbacks);
 
         //modify the checkbox/radio input appearances
-        widget.on('attributeChange', function(data){
-
-            var $checkboxIcons = widget.$container.find('.real-label > span');
+        widget.on('attributeChange', function (data) {
+            const $checkboxIcons = widget.$container.find('.real-label > span');
 
-            if(data.element.serial === interaction.serial && data.key === 'maxChoices'){
-                if(parseInt(data.value) === 1){
+            if (data.element.serial === interaction.serial && data.key === 'maxChoices') {
```

Comment:
There is an issue here. If I uncheck the `Max` constraint, it returns a value of `1` even is disabled. If I want to set a `Min` of 2 but with no `Max`, this is a problem. In this case, I have radio buttons too, but I want to check more than 1 option.
In the Mockup, this case should be a checkbox, not a radio button.
Can you please take a look?

Reply:
@marpesia updated, please check

Reply:
It is working with the new code, but I found a new issue.
If you disabled the checkbox, it is still possible to change the Min/Max numbers and I've checked that it works.
![image](https://user-images.githubusercontent.com/11692751/158975368-1a019a5d-2ea8-40c9-aa33-753bd0318746.png)
And checking Figma I realized that we need to add a margin/padding on the left side
![image](https://user-images.githubusercontent.com/11692751/158976182-a6f9bebc-9ef8-4b3f-9957-4316854ee88c.png)

Reply:
_If you disabled the checkbox, it is still possible to change the Min/Max numbers and I've checked that it works._ - this is right behaviour
Styes are fixed